### DISCUSSION TITLE
How to create foreign keys with custom names

### DISCUSSION THREAD

**Discussion by nizamani** [hasura/graphql-engine#7414] [Question] [answered]
Hi Hasura Team,
We are using Hasura CLI, we make changes locally and then they are deployed to cloud through github. Question I want to ask is this. Is there is any way we can rename foreign key constraints to custom names? Just like check constraints we have name them with any custom names we like it would be nice if we can do the same for foreign keys. Right now Hasura assigns name to foreign keys automatically and I don't see an option in Hasura cli console to name foreign keys.

**Comment by rikinsk** [comment, marked as answer]
@nizamani This is currently not possible to do while creating a foreign key. Shall take this as a feature request.

In the meanwhile, you can rename the created foreign key form the `SQL` tab of the console by running the following query:

`ALTER TABLE "<name>" RENAME CONSTRAINT "<old_fkey_name>" TO "<new_fkey_name>";`

**Reply by nizamani** [reply]
Thank you for your response and taking this as a feature request. This will definitely be a nice feature to have specially for companies that have standards for naming their constraints.
